module-system/sov-modules-api/src/default_signature.rs

```diff
@@ -181,3 +183,25 @@ fn map_error(e: ed25519_dalek::SignatureError) -> std::io::Error {
 fn map_error(_e: ed25519_dalek::SignatureError) -> std::io::Error {
     std::io::Error::new(std::io::ErrorKind::Other, "Signature error")
 }
+
+impl FromStr for DefaultPublicKey {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let bytes = hex::decode(s)?;
+        let pub_key = DalekPublicKey::from_bytes(&bytes)
+            .map_err(|_| anyhow::anyhow!("Invalid public key"))?;
+        Ok(DefaultPublicKey { pub_key })
+    }
+}
+
+impl FromStr for DefaultSignature {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let bytes = hex::decode(s)?;
+        let msg_sig =
+            DalekSignature::from_bytes(&bytes).map_err(|_| anyhow::anyhow!("Invalid signature"))?;
```

Comment:
Having source error to be included at least in form of text can help debugging
```suggestion
            DalekSignature::from_bytes(&bytes).map_err(|e| anyhow::anyhow!("Invalid signature: {:?}", e))?;
```